lib/acl.c

```diff
@@ -73,6 +52,315 @@ drop_unmatched_ipv6_pkts(struct rte_mbuf **pkts, unsigned int num_pkts,
 	return 0;
 }
 
+int
+process_acl(struct gatekeeper_if *iface, unsigned int lcore_id,
+	struct acl_search *acl, struct acl_state *astate)
+{
+	struct rte_mbuf *pkts[astate->func_count][GATEKEEPER_MAX_PKT_BURST];
+	int num_pkts[astate->func_count];
+	unsigned int socket_id = rte_lcore_to_socket_id(lcore_id);
+	const char *proto_name;
+	unsigned int i;
+	int ret;
+
+	switch (astate->proto) {
+	case ETHER_TYPE_IPv4:
```

Comment:
This is the only place at which field `proto` of `struct acl_state` is used. And it has led to callers of `process_pkts_acl()` to write something like `&gt_conf->net->front.ipv4_acls` every time.

My solution to those problems:
1. Drop field `proto` of `struct acl_state`;
2. Replacing parameter `astate` of `process_pkts_acl()` with the protocol;
3. `process_pkts_acl()` should own this `switch...case` statement, and pass parameters`astate`, `proto_if_configured`, and `proto_name` to `process_acl()`.
 So passing the protocol instead of parameter `astate` would enable us to drop field `proto`. Moreover, this would simplify the call of `process_pkts_acl`, which now requires the caller to write something like `&gt_conf->net->front.ipv4_acls` every time.

Reply:
Since `process_pkts_acl()` is inline and its parameter protocol is often going to be a constant, the compiler will eliminate the `switch...case` at compile time. A small bonus of the proposed solution.